README.rst

```diff
@@ -1,7 +1,202 @@
+priority. If some of them are not specified, then suitable environment
+variables are used, otherwise the default values are used.
+
+The ``process_env`` gets environment variables, otherwise default values are
+used.
+
+Two of these functions accept the first position argument which is used as
+a prefix for environment variables, e.g. if ``"CEAGLE"`` was specified, then
+``CEAGLE_CONF`` will be expected as an environment variable.
+
+The full list of supported command line arguments and environment variables:
+
+=================  ====================  =======  =====================================
+Argument           Environment variable  Default  Description
+=================  ====================  =======  =====================================
+--debug            <SERVICE>_DEBUG       false    Use DEBUG instead of INFO for
+                                                  logging, possible values true/false.
+--config-file      <SERVICE>_CONF                 Path to a YAML-configuration file.
+--log-config-file  <SERVICE>_LOG_CONF             Path to a file with configuration for
+                                                  `Python logging module`_
+=================  ====================  =======  =====================================
+
+.. _Python logging module: https://docs.python.org/3/library/logging.config.html#configuration-file-format
+
+Both functions support the default location of a configuration file in case if
+it was not specified through ``--config-file`` or ``<SERVICE>_CONF``.
+The default location can be set using the ``default_config_path`` parameter and
+it will be used only if this file exists.
+
+The list of examples to understand priorities how a configuration file is
+choosen:
+
+================  ================  ===================  =================
+--config-file     <SERVICE>_CONF    default_config_path  Result
+================  ================  ===================  =================
+/etc/ceagle.yaml  /etc/config.yaml  /etc/default.yaml    /etc/ceagle.yaml
+<not set>         /etc/config.yaml  /etc/default.yaml    /etc/config.yaml
+<not set>         <not set>         /etc/default.yaml    /etc/default.yaml
+                                    (exists)
+<not set>         <not set>         /etc/default.yaml    <not set>
+                                    (does not exist)
+================  ================  ===================  =================
+
+Validation and Defaults
+~~~~~~~~~~~~~~~~~~~~~~~
+
+By default ``oss_lib.config`` expects that all configuration settings pass
+through validation in the JSON Schema-like format.
+The ``validation_schema`` parameter expects a dict which populates only
+the ``properties`` parameter in the schema. For example, if your application
+expects two top defined parameters in :
+
+.. code-block:: python
+
+      SCHEMA = {
+          "backend": {
+              "type": "object",
+              "properties": {
+                  "elastic": {
```

Comment:
It would be nice to have realistic example - if I'm not mistaken, we had discussed backend configuration structure to be:

    {
        "type": "<here type e.g. elastic>",
        "connection": <here backend-related format, e.g. list or str>
    }